test/app/basic-auth.js

```diff
@@ -1,18 +1,20 @@
+const koala = require('../../lib');
+const request = require('supertest');
 
-describe('Basic Auth', function () {
-  it('should return the value', function (done) {
-    var app = koala()
-    app.use(function* (next) {
-      this.body = this.request.basicAuth
-    })
+describe('Basic Auth', () => {
+  it('should return the value', done => {
```

Comment:
Warning, Mocha discourages arrows in example functions because it breaks the `this` context. If we aren't using that anywhere I'm fine with this, though it could potentially be a little confusing.

Reply:
@nickmccurdy Well... as was the case with the all the `var` to `let`, this was automatically fixed by `eslint` (with its `--fix` option). I agree with [your comment](https://mochajs.org/#arrow-functions), but I'd say we leave it like this for the time being, based on:

- We will be replacing `mocha` soon :)
- The `this` binding is not that frequently needed, in my experience (we are not using it in our tests). And in those specific cases, we can always change the `=>` into a regular `function`.

Reply:
Yea I've been using Jest instead lately, and it doesn't even support binding to `this`. I've grown to like it though, with all the new block scoping syntax it's easier to simply override variables than have the boilerplate of putting `this` assignments in `before` blocks. Agree to leaving it as is, I was just saying we should keep it consistent until we switch away from Mocha